### PR TITLE
Fix false positives for non-CSS blocks in max-empty-lines

### LINKED ISSUE #3221: Fix false positives in max-empty-lines
In the past week, Stylelint has been flagging "errors" in my HTML files.  Of course, it should not be checking HTML at all (there is no CSS in these files, they are simple Angular templates).

Things like  "string-no-newline" and "max-empty-lines".  I _do_ want these rules enabled for CSS/SCSS, but not in HTML.

I updated the local copy of stylelint in my project to 9.1.3 and prettier-stylelint to 0.4.2, but this was happening before these updates.

```
<div class="home-hero-main">

  
</div>




<div class="container">
  <div class="row">
    <div class="col-md-12">

    </div>

  </div>
</div>
```

VS Code version: Code - Insiders 1.22.0-insider (509b8084d46b82ec5851447903cebd914e4535a5, 2018-03-15T09:25:17.428Z)
OS version: Darwin x64 17.4.0

<details><summary>Extensions (17)</summary>

Extension|Author (truncated)|Version
---|---|---
vsc-material-theme|Equ|1.5.1
code-settings-sync|Sha|2.9.0
Bookmarks|ale|0.18.2
project-manager|ale|0.24.1
vscode-eslint|dba|1.4.7
tslint|eg2|1.0.28
prettier-vscode|esb|1.2.2
auto-rename-tag|for|0.0.15
path-autocomplete|ion|1.8.1
svg|joc|0.0.9
ecdc|mit|0.11.0
HTMLHint|mka|0.4.0
vscode-scss|mrm|0.6.2
vscode-stylefmt|mrm|2.5.0
vscode-versionlens|pfl|0.20.0
stylelint|shi|0.34.2
rewrap|stk|1.8.0

</details><details><summary>Results</summary>

Setting|Extension|Score
---|---|---
stylefmt.configBasedir|mrmlnc.vscode-stylefmt|0.018
stylefmt.config|mrmlnc.vscode-stylefmt|0.017
stylefmt.useStylelintConfigOverrides|mrmlnc.vscode-stylefmt|0.017
stylefmt.showErrorMessages|mrmlnc.vscode-stylefmt|0.001

</details>

### PR DESCRIPTION
**Pull request by gucong3000**
> Which issue, if any, is this issue related to?

Closes #3221

> Is there anything in the PR that needs further explanation?

e.g. "No, it's self explanatory."

### CODE REVIEW

**Comment by hudochenkov** [comment]
Would you mind running `npm run prettier:fix`, please? Or rebase your branch on `master` as it contains a fix for Prettier #3232.

**Comment by gucong3000** [comment]
> If this move away from style-search

Yes.

> Would you mind running npm run prettier:fix, please?

Done.

File: lib/rules/max-empty-lines/index.js [hunk 1]
```diff
@@ -37,14 +37,15 @@ const rule = function(max, options) {
...
 
-    const rootString = root.toString();
     const repeatLFNewLines = _.repeat("\n", maxAdjacentNewlines);
     const repeatCRLFNewLines = _.repeat("\r\n", maxAdjacentNewlines);
     const ignoreComments = optionsMatches(options, "ignore", "comments");
 
-    styleSearch({ source: rootString, target: "\n" }, match => {
-      checkMatch(rootString, match.endIndex, root);
-    });
+    if (root.constructor.name === "Document") {
```
Comment (hudochenkov) [inline lib/rules/max-empty-lines/index.js]: It's not a part of PostCSS, isn't it? Could you add a comment to the code explaining this `if` condition and what it `Document`, please?

**Comment by hudochenkov** [comment]
@gucong3000 little tip: don't rebase PR without much need, if other people made review and requested changes. Rebase makes impossible to see what changes were made since last review. And we have to review whole thing again. “Squash and merge” will make history clean anyway.

**Comment by gucong3000** [comment]
@hudochenkov Thank you for reminding me.

**Comment by jeddy3** [comment]
> -   Fixed: `max-empty-lines` false positives for non-CSS blocks ([#3229](https://github.com/stylelint/stylelint/pull/3229)).

**Pull request merged**